### PR TITLE
fix: treat gpt-6 names as the gpt-5 request family in OpenAI and Azure configs

### PR DESCRIPTION
**Pull request by mateo-berri**
## TLDR

Problem this solves:

- gpt-6-astra fails every `"gpt-5" in model` check, so it gets plain chat handling
- Chat completions 400 on `reasoning_effort`, `max_tokens`, and function tools
- Responses and Messages forward `temperature`, which OpenAI rejects

How it solves it:

- One helper matches gpt-5 and gpt-6 names, still excluding gpt-5-chat
- OpenAI chat, OpenAI Responses, and both Azure configs use it
- gpt-6 names count as gpt-5.4+ for the tools bridge

## User Flow

Before: a developer pointing an app at gpt-6-astra gets a 400 on request shapes that already work for gpt-5.5

1. They POST https://litellm-domain/v1/chat/completions with `"max_tokens": 200` and `"reasoning_effort": "low"` and get a 400 saying `openai does not support parameters: ['reasoning_effort']`
2. They POST https://litellm-domain/v1/chat/completions with a `get_weather` function tool and no reasoning_effort and get a 400 from OpenAI saying function tools with reasoning_effort are not supported in /v1/chat/completions
3. They POST https://litellm-domain/v1/responses with `"temperature": 0` and `"drop_params": true` and get a 400 from OpenAI saying `'temperature' is not supported with this model`
4. They POST https://litellm-domain/v1/messages with the same temperature and drop_params and get the same 400

After: the same four requests return 200

1. They POST https://litellm-domain/v1/chat/completions with `"max_tokens": 200` and `"reasoning_effort": "low"` and get a 200 with content `hello`
2. They POST https://litellm-domain/v1/chat/completions with the `get_weather` function tool and no reasoning_effort and get a 200 carrying a `get_weather` tool call for Paris
3. They POST https://litellm-domain/v1/responses with `"temperature": 0` and `"drop_params": true` and get a 200 with output text `hello`
4. They POST https://litellm-domain/v1/messages with the same temperature and drop_params and get a 200 with text `hello`

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] The handful of test files covering my change pass locally, e.g. `uv run pytest tests/test_litellm/<your_test_file>.py -v`. Leave the suites (`make test-unit-*`, `make test-unit`) to CI: it finishes in ~15 minutes where a laptop takes an hour or more
- [ ] My PR passes all required CI/CD checks (e.g., lint, schema.d.ts sync check, etc.)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [ ] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review (Greptile reviews automatically once the PR is opened; only comment `@greptileai` to re-request a review after pushing changes)

## Delays in PR merge?

If you're seeing a delay in your PR being merged, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA).

## Screenshots / Proof of Fix

Before is the merge base, which already carries the gpt-6-astra price map from #39622, so both runs price the same way. Every call below hits OpenAI's real gpt-6-astra

Shared setup:

`astra_config.yaml`

```yaml
model_list:
  - model_name: gpt-6-astra
    litellm_params:
      model: openai/gpt-6-astra
      api_key: os.environ/OPENAI_API_KEY

general_settings:
  master_key: sk-1234
```

`chat_params.json`

```json
{"model": "gpt-6-astra", "messages": [{"role": "user", "content": "Reply with one word: hello"}], "max_tokens": 200, "reasoning_effort": "low"}
```

`chat_tools.json`

```json
{"model": "gpt-6-astra", "messages": [{"role": "user", "content": "What is the weather in Paris? Use the tool."}],
 "tools": [{"type": "function", "function": {"name": "get_weather", "parameters": {"type": "object", "properties": {"city": {"type": "string"}}, "required": ["city"]}}}]}
```

`responses_temperature.json`

```json
{"model": "gpt-6-astra", "input": "Reply with one word: hello", "temperature": 0, "drop_params": true}
```

`messages_temperature.json`

```json
{"model": "gpt-6-astra", "max_tokens": 200, "messages": [{"role": "user", "content": "Reply with one word: hello"}], "temperature": 0, "drop_params": true}
```

Proxy:

```bash
LITELLM_LOCAL_MODEL_COST_MAP=True python litellm/proxy/proxy_cli.py --config astra_config.yaml --port 4000 --use_v2_migration_resolver
```

#### Chat completions with max_tokens and reasoning_effort

1. `curl -s http://localhost:4000/v1/chat/completions -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @chat_params.json`

```json
{"error": {"message": "litellm.UnsupportedParamsError: openai does not support parameters: ['reasoning_effort'], for model=gpt-6-astra. To drop these, set `litellm.drop_params=True` or for proxy:\n\n`litellm_settings:\n drop_params: true`\n. \n If you want to use these params dynamically send allowed_openai_params=['reasoning_effort'] in your request.. Received Model Group=gpt-6-astra\nAvailable Model Group Fallbacks=None", "type": "None", "param": null, "code": "400"}}
```

#### Chat completions with a function tool and default reasoning

1. `curl -s http://localhost:4000/v1/chat/completions -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @chat_tools.json`

```json
{"error": {"message": "litellm.BadRequestError: OpenAIException - Function tools with reasoning_effort are not supported for gpt-6-astra in /v1/chat/completions. To use function tools, use /v1/responses or set reasoning_effort to 'none'.. Received Model Group=gpt-6-astra\nAvailable Model Group Fallbacks=None", "type": "invalid_request_error", "param": "reasoning_effort", "code": "400"}}
```

#### Responses with temperature

1. `curl -s http://localhost:4000/v1/responses -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @responses_temperature.json`

```json
{"error": {"message": "litellm.BadRequestError: OpenAIException - {\n  \"error\": {\n    \"message\": \"Unsupported parameter: 'temperature' is not supported with this model.\",\n    \"type\": \"invalid_request_error\",\n    \"param\": \"temperature\",\n    \"code\": null\n  }\n}. Received Model Group=gpt-6-astra\nAvailable Model Group Fallbacks=None", "type": null, "param": null, "code": "400"}}
```

#### Messages with temperature

1. `curl -s http://localhost:4000/v1/messages -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @messages_temperature.json`

```json
{"error": {"message": "litellm.BadRequestError: OpenAIException - {\n  \"error\": {\n    \"message\": \"Unsupported parameter: 'temperature' is not supported with this model.\",\n    \"type\": \"invalid_request_error\",\n    \"param\": \"temperature\",\n    \"code\": null\n  }\n}. Received Model Group=gpt-6-astra\nAvailable Model Group Fallbacks=None", "type": null, "param": null, "code": "400"}}
```

#### Chat completions with max_tokens and reasoning_effort

1. `curl -s http://localhost:4000/v1/chat/completions -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @chat_params.json` (id, model, usage, and message fields shown)

```json
{
  "id": "chatcmpl-EK8JFSLZSEN09MmRMrHufA4C2577E",
  "model": "gpt-6-astra",
  "usage": {"completion_tokens": 4, "prompt_tokens": 12, "total_tokens": 16},
  "content": "hello",
  "tool_calls": null
}
```

#### Chat completions with a function tool and default reasoning

1. `curl -s http://localhost:4000/v1/chat/completions -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @chat_tools.json` (id, model, usage, and message fields shown)

```json
{
  "id": "chatcmpl-f2362822-b25e-460b-b2fe-3a3006583dce",
  "model": "gpt-6-astra",
  "usage": {"completion_tokens": 18, "prompt_tokens": 46, "total_tokens": 64},
  "content": null,
  "tool_calls": [
    {
      "index": 0,
      "function": {"arguments": "{\"city\":\"Paris\"}", "name": "get_weather"},
      "id": "call_skQlKBcCP2TNb1TmSlab3lM8",
      "type": "function"
    }
  ]
}
```

#### Responses with temperature

1. `curl -s http://localhost:4000/v1/responses -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @responses_temperature.json` (id, model, usage, and output text shown)

```json
{
  "id": "resp_ydlZiGvaC8p71CLgwWGaj_ebwNZbKvZ-lm79ow2GrK6pV7U8tGvIhkPJGzydDVE4nZ_MXs_2JTv_oI-Mgcnw17lfpxRSyvK87wKWRibaDuqNpLk0nyDvCTMyK4To_Vk6stB7x--Td2f6yeM8_gOUltouW_WzuvlzZLlHfDqVaU5sAR2L8Vni0fUjc9je0BhJwlGpOHt5FhV8sv6aHSNwnTC3EzZujGrrVDVc9TcN7IB2ZPMceZggwHX9iT39ew5CVAKHKYXijNwLXLkukrMRDTkTarST9_n233RT5nvcAVCBfLmIhg-wFLGBGv_vZ-76h2T7vsXx7rORySuwt8kIJxP8Ut7zqcRuPepbKCBf9j1PqDS9kT3ZUJHQKANvb7hUPARBX3gBtIYQw9bvAbOX9ElX0pPDydjPpxQ_pl-10dNtHAQXkzRxIyNijg-vWMZNSr2EShOme_anrZ4L-KYlBSxs",
  "model": "gpt-6-astra",
  "usage": {"input_tokens": 12, "output_tokens": 5, "total_tokens": 17},
  "output": [{"type": "message", "text": "hello"}]
}
```

#### Messages with temperature

1. `curl -s http://localhost:4000/v1/messages -H 'Authorization: Bearer sk-1234' -H 'Content-Type: application/json' -d @messages_temperature.json` (id, model, usage, and content shown)

```json
{
  "id": "resp_bGl0ZWxsbTpjdXN0b21fbGxtX3Byb3ZpZGVyOm9wZW5haTttb2RlbF9pZDozMDdiOWQxZjliZDVlODY5ZjIzOWUwNzg1ZWY1YTNiMWYyN2YyNmZhMTgzYmVmZjZkOWM4MDgxZTAyOTVkOTQ5O3Jlc3BvbnNlX2lkOnJlc3BfMGM4YjZhMDc2ODEyZjRjYjAwNmE5OWQ4MGQ4NGQ4ODdkMDhlZTJjMzMxZWVhOWQ0YjA=",
  "model": "gpt-6-astra",
  "usage": {"input_tokens": 12, "output_tokens": 5},
  "content": [{"type": "text", "text": "hello"}]
}
```

## Type

🐛 Bug Fix

### Low

- OpenAI's chat completions rejects `reasoning_effort: max` for gpt-6-astra (low through xhigh only)
  - The Responses API accepts max, verified live, so the price map's max flag stands
- Bedrock Converse keeps its `openai.gpt-5` checks until the Bedrock model id for Astra is public

## Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: litellm/llms/openai/chat/gpt_5_transformation.py [hunk 1]
```diff
@@ -61,6 +61,14 @@ def _get_effort_level(value: str | dict | None) -> str | None:
     return None
 
 
+GPT_REASONING_SERIES_MARKERS: Final = ("gpt-5", "gpt-6")
+
+
+def is_gpt_reasoning_series_name(model: str) -> bool:
+    normalized: Final = model.split("/")[-1]
+    return any(marker in model for marker in GPT_REASONING_SERIES_MARKERS) and not normalized.startswith("gpt-5-chat")
```
[the remark this answers is not in the document]
Comment (mateo-berri) [reply litellm/llms/openai/chat/gpt_5_transformation.py:69]: OK, fair but it's best to refactor in a follow up. This is blocking the nightly

**Comment by mateo-berri** [comment]
@greptileai

**Review by yuneng-berri — APPROVED** [review]
